packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -153,20 +163,26 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	 */
 	private readonly initialTransactionRevToRebasedRev = new Map<RevisionTag, RevisionTag>();
 	private disposed = false;
+	private readonly unsubscribeBranchTrimmer?: () => void;
 	/**
 	 * Construct a new branch.
 	 * @param head - the head of the branch
 	 * @param changeFamily - determines the set of changes that this branch can commit
+	 * @param branchTrimmer - an optional event emitter that informs the branch it has been trimmed
```

Comment:
I think we should add a clause here that says "If this is not supplied, then the branch must never be trimmed". We should probably also explain what trimming is (a very specific kind of mutation which is the only allowed mutation). You could do that in the doc comment for `BranchTrimmingEvents`and link to it here if you think that'd be more centralized. Now that "trimming" is leaking out of EditManager, we should have some place that's not too hard to find which explains what it's all about, without someone having to dig into EditManager internals.